spark-operator/src/main/java/org/apache/spark/k8s/operator/config/SparkOperatorConf.java

```diff
@@ -0,0 +1,463 @@
+      ConfigOption.<Integer>builder()
+          .key("spark.kubernetes.operator.rateLimiter.refreshPeriodSeconds")
+          .enableDynamicOverride(false)
+          .description("Operator rate limiter refresh period(in seconds) for each resource.")
+          .typeParameterClass(Integer.class)
+          .defaultValue(15)
+          .build();
+
+  public static final ConfigOption<Integer> RATE_LIMITER_LIMIT_FOR_PERIOD =
+      ConfigOption.<Integer>builder()
+          .key("spark.kubernetes.operator.rateLimiter.limitForPeriod")
+          .enableDynamicOverride(false)
+          .description(
+              "Max number of reconcile loops triggered within the rate limiter refresh "
+                  + "period for each resource. Setting the limit <= 0 disables the "
+                  + "limiter.")
+          .typeParameterClass(Integer.class)
+          .defaultValue(5)
+          .build();
+
+  public static final ConfigOption<Integer> RETRY_INITIAL_INTERVAL_SECONDS =
+      ConfigOption.<Integer>builder()
+          .key("spark.kubernetes.operator.retry.initialIntervalSeconds")
+          .enableDynamicOverride(false)
+          .description("Initial interval(in seconds) of retries on unhandled controller errors.")
+          .typeParameterClass(Integer.class)
+          .defaultValue(5)
+          .build();
+
+  public static final ConfigOption<Double> RETRY_INTERVAL_MULTIPLIER =
+      ConfigOption.<Double>builder()
+          .key("spark.kubernetes.operator.retry.intervalMultiplier")
+          .enableDynamicOverride(false)
+          .description(
+              "Interval multiplier of retries on unhandled controller errors. Setting "
+                  + "this to 1 for linear retry.")
+          .typeParameterClass(Double.class)
+          .defaultValue(1.5)
+          .build();
+
+  public static final ConfigOption<Integer> RETRY_MAX_INTERVAL_SECONDS =
+      ConfigOption.<Integer>builder()
+          .key("spark.kubernetes.operator.retry.maxIntervalSeconds")
+          .enableDynamicOverride(false)
+          .description(
+              "Max interval(in seconds) of retries on unhandled controller errors. "
+                  + "Set to non-positive for unlimited.")
+          .typeParameterClass(Integer.class)
+          .defaultValue(-1)
+          .build();
+
+  public static final ConfigOption<Integer> RETRY_MAX_ATTEMPTS =
+      ConfigOption.<Integer>builder()
+          .key("spark.kubernetes.operator.retry.maxAttempts")
+          .enableDynamicOverride(false)
+          .description(
+              "Max attempts of retries on unhandled controller errors. Setting this to "
+                  + "non-positive value means no retry.")
+          .typeParameterClass(Integer.class)
+          .defaultValue(15)
+          .build();
+
+  public static final ConfigOption<Long> RETRY_ATTEMPT_AFTER_SECONDS =
+      ConfigOption.<Long>builder()
+          .key("spark.kubernetes.operator.retry.attemptAfterSeconds")
```

Comment:
Could you clarify
- When `initialIntervalSeconds` and `intervalMultiplier` is used
- When `attemptAfterSeconds` is used

Reply:
`When initialIntervalSeconds and intervalMultiplier is used` - they are for reconciler behavior

`When attemptAfterSeconds is used` - this is for requests made to API server. 

To avoid further confusion, lemme use another namespace for the latter.